starport-core/src/main/scala/com/krux/starport/metric/CloudWatchReporter.scala

```diff
@@ -0,0 +1,925 @@
+        .withMaximum(convertDuration(snapshot.getMax))
+
+      val dimensions: Set[Dimension] =
+        new LinkedHashSet[Dimension](builder.globalDimensions)
+      dimensions.add(
+        new Dimension()
+          .withName(DIMENSION_NAME_TYPE)
+          .withValue(DIMENSION_SNAPSHOT_SUMMARY)
+      )
+
+      dimensions.addAll(dimensionedName.getDimensions.asJavaCollection)
+      metricData.add(
+        new MetricDatum()
+          .withTimestamp(new Date(builder.clock.getTime))
+          .withMetricName(dimensionedName.name)
+          .withDimensions(dimensions)
+          .withStatisticValues(statisticSet)
+          .withStorageResolution(
+            if (highResolution) HIGH_RESOLUTION
+            else STANDARD_RESOLUTION
+          )
+          .withUnit(standardUnit)
+      )
+
+    }
+  }
+
+  private def stageMetricDatumWithRawSnapshot(
+    metricConfigured: Boolean,
+    metricName: String,
+    snapshot: Snapshot,
+    standardUnit: StandardUnit,
+    metricData: List[MetricDatum]
+  ): Unit = {
+    if (metricConfigured) {
+      val dimensionedName: DimensionedName = DimensionedName.decode(metricName)
+      val total: Double = snapshot.getValues.toStream.sum
+
+      val statisticSet: StatisticSet = new StatisticSet()
+        .withSum(total)
+        .withSampleCount(snapshot.size.toDouble)
+        .withMinimum(snapshot.getMin.toDouble)
+        .withMaximum(snapshot.getMax.toDouble)
+
+      val dimensions: Set[Dimension] = new LinkedHashSet[Dimension](builder.globalDimensions)
+
+      dimensions.add(
+        new Dimension()
+          .withName(DIMENSION_NAME_TYPE)
+          .withValue(DIMENSION_SNAPSHOT_SUMMARY)
+      )
+
+      dimensions.addAll(dimensionedName.getDimensions().asJavaCollection)
+
+      metricData.add(
+        new MetricDatum()
+          .withTimestamp(new Date(builder.clock.getTime))
+          .withMetricName(dimensionedName.name)
+          .withDimensions(dimensions)
+          .withStatisticValues(statisticSet)
+          .withStorageResolution(if (highResolution) HIGH_RESOLUTION else STANDARD_RESOLUTION)
+          .withUnit(standardUnit)
+      )
+    }
+  }
+
+  private def cleanMetricValue(metricValue: Double): Double = {
+    val absoluteValue: Double = Math.abs(metricValue)
+    if (absoluteValue < SMALLEST_SENDABLE_VALUE) {
+// Allow 0 through untouched, everything else gets rounded to SMALLEST_SENDABLE_VALUE
+      if (absoluteValue > 0) {
+        if (metricValue < 0) {
+          -SMALLEST_SENDABLE_VALUE
+        } else {
+          SMALLEST_SENDABLE_VALUE
+        }
+      }
+    } else if (absoluteValue > LARGEST_SENDABLE_VALUE) {
+      if (metricValue < 0) {
+        -LARGEST_SENDABLE_VALUE
+      } else {
+        LARGEST_SENDABLE_VALUE
+      }
+    }
+    metricValue
+  }
+
+}
+
+object CloudWatchReporter {
+
+  private val LOGGER: Logger = LoggerFactory.getLogger(classOf[CloudWatchReporter])
+
+  // Visible for testing
+  val DIMENSION_NAME_TYPE: String = "Type"
```

Comment:
let's use `CamelCase` if we would like these to be scala

Reply:
Done.